CONVENTIONS.md

```diff
@@ -0,0 +1,95 @@
+## Plugin Naming Conventions
+
+These are _recommended_ plugin naming conventions. YMMV.
+
+- [GitHub](#github)
+    - [OpenSearch Plugins](#opensearch-plugins)
+    - [OpenSearch Dashboard Plugins](#opensearch-dashboard-plugins)
+- [Metadata](#metadata)
+    - [OpenSearch Plugins](#opensearch-plugins)
+    - [OpenSearch Dashboard Plugins](#opensearch-dashboard-plugins)
+- [Artifacts](#artifacts)
+- [Folders](#folders)
+- [Classes](#classes)
+- [APIs](#apis)
+- [Indices](#indices)
+- [Identifiers](#identifiers)
+- [Variables](#variables)
+
+### GitHub
+
+- Within the [opensearch-project org](https://github.com/opensearch-project), do not include the word `OpenSearch` or `OpenSearch Dashboards` in the repo name.
+- Inherit templates, CoC, etc., from [opensearch-project/.github](https://github.com/opensearch-project/.github).
+- Use lowercase repo names.
+- Provide a meaningful description, e.g. `An OpenSearch Dashboards plugin to perform real-time and historical anomaly detection on OpenSearch data.`
+
+#### OpenSearch Plugins
+
+- Do not include the word `plugin` in the repo name, e.g. [job-scheduler](https://github.com/opensearch-project/job-scheduler).
+
+#### OpenSearch Dashboard Plugins
+
+- Prefix OpenSearch Dashboards plugins with `dashboards-`, e.g. [dashboards-reports](https://github.com/opensearch-project/dashboards-reports).
+
+### Metadata
+
+#### OpenSearch Plugins
+
+- Plugin name starts with `opensearch-`.
+- Include `opensearch-` in the plugin name.
+- Description does not end with a period.
+- Include `OpenSearch` and `plugin` in the descripion. 
+
+```groovy
+opensearchplugin {
+    name 'opensearch-job-scheduler'
+    description 'OpenSearch Job Scheduler plugin'
+    classname 'com.amazon.opensearch.jobscheduler.JobSchedulerPlugin'
+}
+```
+
+#### OpenSearch Dashboard Plugins
+
+- Plugin ID is lowercase camelcase, e.g. `notebooksDashboards`.
+- Version follows [semver](https://semver.org/).
+
+```json
```

Comment:
May be clarify which file it is this part of? Or probably link to an example plugin? WDYT?